source/TimePicker.js

```diff
@@ -585,14 +608,27 @@
 		meridiemPickerChanged: function (sender, event) {
 			if(this.syncingPickers) return true;
 
-			var hour = this.$.hour.get('value'),
-				value = event.value;
+			var meridiems = this.meridiems,
+				hour = this.$.hour.getValue(),
+				oldIndex = event.old,
+				newIndex = this.$.meridiem.getValue(),
+				offset, start, end, newHour;
 
-			if (this.value) {
-				// value is 0 for am, 1 for pm
-				// reset the hour to < 12 and then add 12 if it's pm
-				hour = hour%12 + value*12;
-				this.updateHours(hour);
+			offset = hour - parseInt(meridiems[oldIndex]['start'], 10);
```

Comment:
Just wanted to confirm that the desired behavior is to preserve the offset from the start of the previous meridiem, when changing meridiems. Is this what occurs in other interfaces for this locale?

cc @ehoogerbeets